File: DecisionTreeRegression.R
#Decision Tree Regression 

#regression template 

#polynomial Regression
# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Splitting the dataset into the Training set and Test set
# # install.packages('caTools')
# library(caTools)
# set.seed(123)
# split = sample.split(dataset$Salary, SplitRatio = 2/3)
# training_set = subset(dataset, split == TRUE)
# test_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set = scale(training_set)
# test_set = scale(test_set)


#fitting the regression model to the dataset
#create our regressor
install.packages('rpart')
library(rpart)
regressor = rpart(formula = Salary ~ ., data = dataset,
                  control = rpart.control(minsplit = 1))


#predicting results with regression model
Y_pred = predict(regressor, data.frame(Level = 6.5))







#visualising regresssion results(for higher resolution and smoother curve)
library(ggplot2)
X_grid = seq(min(dataset$Level), max(dataset$level), 0.1)

ggplot() + geom_point(aes(x = dataset$Level, y = dataset$Salary)
                      , color = 'red') +
  geom_line(aes(x = X_grid, y = predict(regressor, newdata = data.frame(Level = X_grid)))
            , color = 'blue') +
  ggtitle('truth or bluff (Decision Tree Regression Model)') +
  xlab('level') + ylab('salary')

File: DecisionTreeRegression.py
# -*- coding: utf-8 -*-
"""
Created on Sun Nov 19 18:25:45 2017

@author: DIAMONDCORP
"""
#Decision Tree Regression



#importing libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importing datasets

dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
Y = dataset.iloc[:, 2].values

#splitting dataset into training and test set

"""from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.2, 
                      random_state = 0)"""

#feature scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""


#fitting regression model to dataset
#create your regressor here
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor(random_state = 0)
regressor.fit(X, Y)




#predicting a new result with the regression model
Y_pred = regressor.predict(6.5)



#visualising the regression results(for higher resolution and smoother curve)
X_grid = np.arange(min(X), max(X), 0.01)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, Y, color = 'black')
plt.plot(X_grid, regressor.predict(X_grid), color = 'green')
plt.title('truth or bluff (decision tree regression)')
plt.xlabel('position level')
plt.ylabel('salary')
plt.show()




















